pkg/cloudprovider/types.go

```diff
@@ -84,6 +86,7 @@ type InstanceType struct {
 	Overhead *InstanceTypeOverhead
 }
 
+// +k8s:deepcopy-gen=true
```

Comment:
Does it make sense to bring this into the apis package if we're gonna start using it in APIs w/ deepcopy?

Reply:
Not sure what you mean. I figured we could leverage the deepcopy auto-gen from kubebuilder so that I wouldn't have to handroll the deepcopy myself or use a different tool to generate it.

We could feasibly do this for other internal structs, that aren't meant to be externally facing; so, I don't really see the connection between using `deepcopy-gen` and putting it in the apis package.